Doc/library/functions.rst

```diff
@@ -63,17 +63,15 @@ are always available.  They are listed here in alphabetical order.
 
 .. function:: aiter(async_iterable)
 
-   Return an :term:`asynchronous iterator`. This is the async variant
-   of the :func:`iter` builtin, and behaves similarly.
+   Equivalent to calling ``x.__aiter__()``.
 
-   *async_iterable* must be an :term:`asynchronous iterable`,
-   and :func:`aiter` returns an asynchronous iterator for it.
+   ``aiter(aiter(x))`` is the same as ``aiter(x)``.
+   (``aiter(x)`` itself has an ``__aiter__()`` method that returns ``self``.)
 
-   Unlike the :func:`iter` builtin, :func:`aiter` has no 2-argument variant.
-   Often, this variant can be replaced with assignment expressions::
+   Formally, given an :term:`asynchronous iterable`,
+   return an :term:`asynchronous iterator`.
```

Comment:
Much better! Here is how I would rearrange this:
```
Return an :term:`asynchronous iterator`
for an :term:`asynchronous iterable`. 
Equivalent to calling ``x.__aiter__()``.

``aiter(x)`` itself has an ``__aiter__()`` method that returns ``x``,
so ``aiter(aiter(x))`` is the same as ``aiter(x)``.
```

Reply:
Incorporated in the latest revision, thanks!

(((
Come to think of it, `iter` also has the `iter(iter(x)) == iter(x)` property, which is not currently mentioned in [the `iter` docs](https://docs.python.org/3/library/functions.html#iter). The `iter` docs have a lot more work to do though, to cover the 1- and 2-arg variants (and they're already doing this very well).

And now that I'm looking at those again, I notice the only code example there is for the 2-arg variant:
```python
from functools import partial
with open('mydata.db', 'rb') as f:
    for block in iter(partial(f.read, 64), b''):
        process_block(block)
```
If this pattern is now obsoleted by assignment expressions[1], is it worth updating the `iter` docs to (1) mention the `iter(iter(x)) == iter(x)` property, and (2) remove the obsoleted example code? If so, happy to submit a separate PR for that.

[1] as in the following:

```python
with open('mydata.db', 'rb') as f:
    while block := f.read(64):
        process_block(block)
```
)))

Reply:
I don't want to personally get into the weeds about the iter() docs, sorry. Something for the docs WG perhaps.

Reply:
I don't think it's worth updating the docs to promise `iter(iter(x)) == iter(x)` as it isn't a specific benefit to the user beyond logically doing the right thing.

Reply:
I can see that. @gvanrossum, what do you think? I did add a `test_aiter_idempotent()` for this in 6ee8824 to go along with the promise that the docs are now making, but can remove that test along with that part of the docs if that's better.

Reply:
I think there's a different bar for updating the `iter()` docs than for the initial version of the `aiter()` docs. And they don't have to match precisely.

FWIW the main reason the idempotency property is important is because of the implicit `[a]iter()` call in a for-loop, since in
```
for i in x: ...
```
the for-loop calls `iter(x)`, so that in
```
for i in iter(x): ...
```
the for-loop ends up calling `iter(iter(x))`. It's the same for `async for` and `aiter()`.